src/Forms/SearchableDropdownTrait.php

```diff
@@ -0,0 +1,594 @@
+     */
+    public function setSource($source): static
+    {
+        if (!is_a($source, DataList::class)) {
+            throw new InvalidArgumentException('Source must be a DataList');
+        }
+        // Setting to $this->sourceList instead of $this->source because SelectField.source
+        // docblock type is array|ArrayAccess i.e. does not allow DataList
+        $this->sourceList = $source;
+        return $this;
+    }
+
+    /**
+     * Get the field to use for the label of the option
+     *
+     * The default value of 'Title' will map to DataObject::getTitle() if a Title DB field does not exist
+     */
+    public function getLabelField(): string
+    {
+        return $this->labelField;
+    }
+
+    /**
+     * Set the field to use for the label of the option
+     */
+    public function setLabelField(string $labelField): static
+    {
+        $this->labelField = $labelField;
+        return $this;
+    }
+
+    public function getAttributes(): array
+    {
+        $name = $this->getName();
+        if ($this->getIsMultiple() && strpos($name, '[') === false) {
+            $name .= '[]';
+        }
+        return array_merge(
+            parent::getAttributes(),
+            [
+                'name' => $name,
+                'data-schema' => json_encode($this->getSchemaData()),
+            ]
+        );
+    }
+
+    /**
+     * Get a list of selected ID's
+     */
+    public function getValueArray(): array
+    {
+        $value = $this->Value();
+        if (empty($value)) {
+            return [];
+        }
+        if (is_array($value)) {
+            $arr = $value;
+            // Normalise FormBuilder values to be like Page EditForm values
+            //
+            // Page EditForm $values for non-multi field will be
+            // [
+            //   0 => '10',
+            // ];
+            // FormBuilder $values for non-multi field will be
+            // [
+            //   'label' => 'MyTitle15', 'value' => '10'
+            // ]
+            if (array_key_exists('value', $arr)) {
+                $arr = [$arr['value']];
+            }
+            // Page EditForm $values for multi will be
+            // [
+            //   0 => '10',
+            //   1 => '15'
+            // ];
+            // FormBuilder $values for multi will be
+            // [
+            //   0 => ['label' => '10', 'value' => 'MyTitle10', 'selected' => false],
+            //   1 => ['label' => '15', 'value' => 'MyTitle15', 'selected' => false]
+            // ];
+            if (array_key_exists(0, $arr) && is_array($arr[0]) && array_key_exists('value', $arr[0])) {
+                $arr = array_map(fn ($a) => $a['value'], $arr);
+            }
+            $arr = array_filter($arr);
+            return array_map(fn ($a) => (int) $a, $arr);
+        }
+        try {
+            // things stdClass cannot be cast to string
+            (string) $value;
+        } catch (Error $e) {
+            $value = 0;
```

Comment:
I don't think we should just assume the value is a 0 - throw an exception instead, or just don't catch the error.
When would we be having an `stdClass` passed in here anyway?

Reply:
Should have been "Things like stdClass cannot be cast to string", have updated comment

Not throwing exception as explained here https://github.com/silverstripe/silverstripe-framework/pull/11057#discussion_r1423382356

> When would we be having an stdClass passed in here anyway?

You probably wouldn't, this is just a catch all for any scenarios that have not been considered. We need to cast to string in the block below this so this is just to ensure that we can cast to string

Reply:
We should log a warning here instead then, so developers can still know something went wrong.
This seems to have disappeared though?

Reply:
Have refactored this code out instead.  It would now get logged out at the bottom of this method